From 8168df29488141e4fca1b2c34aa9c22c7d6c92b6 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 11 Jul 2023 18:01:32 +0200
Subject: [PATCH 0129/1695] plat-stm32mp2: add VDERAM SYSCFG support

Adds support for the VDERAM configuration that is present in SYSCFG.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 655b511bd7f91658ee5636c92d9a5d80d614d0ce plat-stm32mp2: add VDERAM SYSCFG support]
Change-Id: I8fc53b1bc937869548311af2e238c94f323deac1
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/stm32_sysconf.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/stm32_sysconf.h b/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
index 1e34a0005..8924ca3e5 100644
--- a/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
+++ b/core/arch/arm/plat-stm32mp2/stm32_sysconf.h
@@ -22,6 +22,17 @@ enum syscon_banks {
 #define SYSCON_ID(bank, offset) (((bank) << 16) | \
 				((offset) & GENMASK_32(15, 0)))
 
+/*
+ * SYSCFG register offsets (base relative)
+ */
+#define SYSCFG_VDERAMCR		SYSCON_ID(SYSCON_SYSCFG, 0x1800U)
+
+/*
+ * SYSCFG_VDERAMCR register offsets
+ */
+#define VDERAMCR_VDERAM_EN		BIT(0)
+#define VDERAMCR_MASK			BIT(0)
+
 /*
  * A35SSC register offsets (base relative)
  */
-- 
2.47.2

